divide_and_conquer/closest_pair_of_points.py

```diff
@@ -1,66 +1,113 @@
 """
-The algorithm finds distance between closest pair of points in the given n points.
-Approach: Divide and conquer 
-The points are sorted based on x-cords 
+The algorithm finds distance btw closest pair of points in the given n points.
+Approach used -> Divide and conquer 
+The points are sorted based on Xco-ords 
 & by applying divide and conquer approach, 
 minimum distance is obtained recursively.
 
-Edge case: closest points lie on different sides of partition
+>> closest points lie on different sides of partition
 This case handled by forming a strip of points 
-which are at a distance (< closest_pair_dis) from mid-point.
-(It is a proven that strip contains at most 6 points)
-And brute force method is applied on strip to find closest points. 
+whose Xco-ords distance is less than closest_pair_dis
+from mid-point's Xco-ords.
+Closest pair distance is found in the strip of points. (closest_in_strip)
 
-Time complexity: O(n * (logn) ^ 2)
+min(closest_pair_dis, closest_in_strip) would be the final answer.
+ 
+Time complexity: O(n * (logn)^2)
 """
 
 
 import math 
 
 
-def euclidean_distance(point1, point2):
-    return math.sqrt(pow(point1[0] - point2[0], 2) + pow(point1[1] - point2[1], 2))
+def euclidean_distance_sqr(point1, point2):
+    return pow(point1[0] - point2[0], 2) + pow(point1[1] - point2[1], 2)
 
 
 def column_based_sort(array, column = 0):
     return sorted(array, key = lambda x: x[column])
     
 
-#brute force approach to find distance between closest pair points
-def dis_btw_closest_pair(points, no_of_points, min_dis = float("inf")):
-    for i in range(no_of_points - 1):
-        for j in range(i+1, no_of_points):
-            current_dis = euclidean_distance(points[i], points[j])
+def dis_btw_closest_pair(points, points_counts, min_dis = float("inf")):
+    """ brute force approach to find distance between closest pair points
+
+    Parameters : 
+    points, points_count, min_dis (tuple(tuple(int, int)), int, int) 
+    
+    Returns : 
+    min_dis (float):  distance between closest pair of points
+
+    """
+
+    for i in range(points_counts - 1):
+        for j in range(i+1, points_counts):
+            current_dis = euclidean_distance_sqr(points[i], points[j])
             if current_dis < min_dis:
                 min_dis = current_dis
     return min_dis
 
 
-#divide and conquer approach
-def closest_pair_of_points(points, no_of_points):
+def dis_btw_closest_in_strip(points, points_counts, min_dis = float("inf")):
```

Comment:
It's not important but if you have a better name for this function, please update it. Long function names are also fine. Using "between" is better than "btw". Also you can use "brute force" word in its name.
These were only some suggestions. It's up to you.